offchain-modules/src/packages/xchain/btc/scripts.ts

```diff
@@ -0,0 +1,233 @@
+import { RPCClient } from 'rpc-bitcoin';
+import bitcore from 'bitcore-lib';
+import { ForceBridgeCore } from '@force-bridge/core';
+import { isEmptyArray } from '@force-bridge/utils';
+const BigNumber = require('bignumber.js');
+
+const BtcLockEventMark = 'ck';
+const BtcUnlockEventMark = 'unlock';
+const CkbBurnSplitMark = 'tx';
+const BtcOPReturnCode = '6a';
+
+export class BTCChain {
+  protected readonly rpcClient: RPCClient;
+  protected readonly multiAddress: string;
+  protected readonly multiPubkeys;
+  protected readonly multiPrivKeys;
+
+  constructor() {
+    const config = ForceBridgeCore.config.btc;
+    const clientParams = config.clientParams;
+    const privKeys = config.privateKeys;
+    this.multiPrivKeys = privKeys.map((pk) => new bitcore.PrivateKey(pk.slice(2)));
+    this.multiPubkeys = this.multiPrivKeys.map((pk) => pk.toPublicKey());
+    const multiAddress = bitcore.Address.createMultisig(this.multiPubkeys, 2, 'testnet').toString();
+    logger.debug(
+      `the multi sign address by calc privkeys is : ${multiAddress}. the provider lock address is ${config.lockAddress}`,
+    );
+    if (multiAddress !== config.lockAddress) {
+      throw new Error(
+        `the multi sign address by calc privkeys is : ${multiAddress}. which different ${config.lockAddress} in conf.`,
+      );
+    }
+    this.multiAddress = multiAddress;
+    this.rpcClient = new RPCClient(clientParams);
+  }
+
+  async watchBtcTxEvents(startHeight = 1, endHeight, handleLockAsyncFunc, handleUnlockAsyncFunc) {
+    for (let blockHeight = startHeight; blockHeight <= endHeight; blockHeight++) {
+      const blockhash = await this.rpcClient.getblockhash({ height: blockHeight });
+      const block = await this.rpcClient.getblock({ blockhash, verbosity: 2 });
+      if (block.tx.length === 1) {
+        continue;
+      }
+      let waitVerifyTxs = block.tx.slice(1);
+      for (let txIndex = 0; txIndex < waitVerifyTxs.length; txIndex++) {
+        const txVouts = waitVerifyTxs[txIndex].vout;
+        const ckbBurnTxHashes = await this.getUnockTxData(waitVerifyTxs[txIndex].vin, txVouts);
+        if (!isEmptyArray(ckbBurnTxHashes)) {
+          logger.debug(
+            'verify for unlock event. block',
+            blockHeight,
+            'tx ',
+            waitVerifyTxs[txIndex].hash,
+            'find ckb burn hashes:',
+            ckbBurnTxHashes,
+          );
+          for (let i = 0; i < ckbBurnTxHashes.length; i++) {
+            await handleUnlockAsyncFunc(ckbBurnTxHashes[i]);
+          }
+        }
+        if (this.isLockTx(txVouts)) {
+          const data: BtcLockData = {
+            blockHeight: block.height,
+            blockHash: block.hash,
+            txId: waitVerifyTxs[txIndex].txid,
+            txHash: waitVerifyTxs[txIndex].hash,
+            rawTx: waitVerifyTxs[txIndex].hex,
+            txIndex: txIndex,
+            amount: new BigNumber(txVouts[0].value).multipliedBy(Math.pow(10, 8)).toString(),
```

Comment:
use es6 [`bigint`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) instead of BigNumber.